### PR TITLE
add option to put toc on cover

### LINKED ISSUE #200: Table of contents on cover sheet
**Is your feature request related to a problem? Please describe.**
Some stakeholders would like the table of contents to be included on the cover sheet, as in previous versions of `gptables`

**Describe the solution you'd like**
An optional parameter for where to place the table of contents when producing or writing a `GPWorkbook`

### PR DESCRIPTION
**Pull request by Sara-Jade-O**
### Proposed Changes
  -

### Related Issues
  - Related to https://github.com/ONSdigital/gptables/issues/200
  - Fixes #

### Pre-requisites
This section may not be fully required if the branch is not merging into main.
Please indicate items that aren't necessary and why, with comments around incomplete checks.

- [ ] Version number has been incremented, according to [SemVer][semver]
- [ ] Changelog has been updated, listing changes to this version. Use the [keep a changelog][changelog] format
- [ ] New features are tested
- [ ] New features follow the Analysis Function Releasing statistics in spreadsheets [guidance][guidance]
- [ ] New features are documented using the [numpydoc][numpy-docstrings] docstring format
- [ ] Other relevant package documentation is updated
- [ ] For new functionality, examples are included in the docs or a [feature request][feature-request] has
been made for it/them
- [ ] Required workflows and pre-commits succeed

[changelog]: [https://keepachangelog.com/en/1.0.0/]
[feature-request]: [https://github.com/best-practice-and-impact/gptables/issues/new?assignees=&labels=enhancement&template=feature_request.md&title=]
[guidance]: https://analysisfunction.civilservice.gov.uk/policy-store/releasing-statistics-in-spreadsheets/
[numpy-docstrings]: [https://numpydoc.readthedocs.io/en/latest/format.html]
[semver]: [https://semver.org/spec/v2.0.0.html]

### CODE REVIEW

File: gptables/core/api.py [hunk 1]
```diff
@@ -69,6 +75,17 @@ def produce_workbook(
     -------
     workbook : gptables.GPWorkbook
     """
+    valid_locations = ("sheet", "cover")
+    if contentsheet_location not in valid_locations:
+        raise ValueError(
+            f"`contentsheet_location` must be one of {valid_locations},"
```
Comment (ChrisSoderberg-ONS) [inline gptables/core/api.py]: @Sara-Jade-O I think this needs to be made explicit in the docstring for `contentsheet_location`, it was a tad confusing for me

File: gptables/test/test_api.py [hunk 2]
```diff
@@ -96,3 +96,99 @@ def test_end_to_end(create_gpworkbook, file_regression):
...
+
+    sheet_names = [ws.get_name() for ws in wb.worksheets()]
+    # Cover sheet should exist, no separate "Table of contents" sheet
+    assert "Cover" in sheet_names
+    assert "Table of contents" not in sheet_names
+    assert "Sheet1" in sheet_names
+    wb.fileclosed = 1
+
+
+def test_contentsheet_location_sheet_default(tmp_path):
```
Comment (ChrisSoderberg-ONS) [inline gptables/test/test_api.py]: @Sara-Jade-O do we need a test case to ensure we can write a content sheet to sheet without a cover since cover is None by default? 
E.g., 
`cover=None`
`contentsheet_location="sheet"`

Comment (ChrisSoderberg-ONS) [reply gptables/test/test_api.py]: Do we also need a test case for `contentsheet_label=None`

File: gptables/test/test_wrappers.py
Comment (ChrisSoderberg-ONS) [inline gptables/test/test_wrappers.py]: @Sara-Jade-O Would you be able to explain these tests for me? Can't figure out what they're doing

**Review by ChrisSoderberg-ONS — CHANGES_REQUESTED** [review]
@Sara-Jade-O Hi Sara-Jade, I've had a look through this, just got a couple of small comments and there is a merge conflict but happy to merge after these are resolved!
